### PR TITLE
Move FFI functions to Zint::Native and improve structs

### PR DESCRIPTION
**Pull request by larskanis**
This moves the FFI functions to the private module `Zint::Native`, to hide it as internal API.

Make use of `Struct.by_ref` to pass and retrieve Struct classes instead of plain pointers. They are type checked that way and it's less ruby code.

Fix memory leak of `Struct::Symbol` and attached structures not being GC'ed.

Add proper methods for retrieval of barcode vector parts instead of a plain `FFI::Struct`.

### CODE REVIEW

File: spec/zint/barcode_spec.rb [hunk 1]
```diff
@@ -132,13 +132,51 @@ module Zint
...
+        expect(strs[0].height).to be > 1.0
+        expect(strs[0].width).to be > 1.0
+        expect(strs[0].colour).to eq(-1)
+      end
+
+      it "exports barcode as vector of circles" do
+        v = described_class.new(value: "ABC").to_vector
+        strs = v.each_circle.to_a
+        expect(strs.size).to eq(0)
+        # TODO use a barcode with circles
```
Comment (larskanis) [inline spec/zint/barcode_spec.rb:172]: @api-walker Which symbol type could be used for testing circles and hexagons?

Comment (api-walker) [reply spec/zint/barcode_spec.rb:172]: Hi Lars,
please have a look at BARCODE_MAXICODE.

It looks like this:
maxicode_2 1

Comment (larskanis) [reply spec/zint/barcode_spec.rb]: OK, thanks! I added specs for hexagons and circles now.

**Comment by larskanis** [comment]
This is IMHO ready to be merged now.

**Pull request merged**